### PR TITLE
Fixes home screen banner issue

### LINKED ISSUE #911: Fix aspect ratio of main banner in home screen
**Is your feature request related to a problem? Please describe.**

when the device is rotated, the main Kiwix banner looks distorted.

Here's the screen:
screenshot_20190109-023355

And heres how it looks when the device is rotated:
screenshot_20190109-024049

**Describe the solution you'd like**

we could create the banner using three layers, (one for the main rectangle base, one for the kiwix logo, and one for the side pattern), instead of one. Then, in rotated mode, we can just change the size of the base layer, and change the relative positions of the side pattern, and the kiwix logo).

### PR DESCRIPTION
**Pull request by sonusourav**
Fixes #911 

Changes: 
- Added separate XML files for both orientations.
- Used different design to make up for the space in landscape orientation.

Screenshots/GIF for the change: 
kiwix_issue_ 911_resolved

### CODE REVIEW

**Comment by sonusourav** [comment]
@siddharth2010  Based on @abdulwd suggestions, I have kept the design as it was earlier. Hope its fine now.

**Comment by siddharth2010** [comment]
Hey could we have something different for night mode

image

I got this by just adding 100% tint and 100% warmth and adding a 10% tint to the image.

**Comment by siddharth2010** [comment]
you could probably experiment with the effects till you get the perfect fit.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 1]
```diff
@@ -106,6 +85,32 @@
...
+import androidx.constraintlayout.widget.Group;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.core.view.GravityCompat;
+import androidx.drawerlayout.widget.DrawerLayout;
+import androidx.recyclerview.widget.ItemTouchHelper;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import butterknife.BindView;
+import butterknife.OnClick;
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: You have not imported Square's code style as described [here](https://github.com/kiwix/kiwix-android/blob/master/docs/codestyle.md) that is why you're getting these changes when you autoformat.

**Comment by sonusourav** [comment]
@siddharth2010 I have made the necessary changes as requested by you and @abdulwd .

File: app/src/main/res/layout/content_main_l.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        android:layout_height="wrap_content"
+        android:id="@+id/constraint_main"
+        android:background="@drawable/back_cover">
+
+      <ImageView
+          android:id="@+id/content_side_image"
+          android:layout_width="0dp"
+          android:layout_height="144dp"
+          android:scaleType="fitEnd"
+          android:contentDescription="@string/side_design"
```
Comment (abdulwd) [inline app/src/main/res/layout/content_main_l.xml]: `@string/side_design` is assigned `sideDesign`, as the `contentDescription is used by screen readers you should either give it a meaningful string that can we be understood by a person when it is read to him/her or you should set it to "@null" for elements that are only for decoration purposes. You should read this full heading: https://developer.android.com/guide/topics/ui/accessibility/apps#label-elements. In this case, you must use "@null".

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java [hunk 3]
```diff
@@ -106,11 +112,34 @@ public void onPageFinished(WebView view, String url) {
 
   private void inflateHomeView(WebView view) {
     LayoutInflater inflater = LayoutInflater.from(view.getContext());
-    home = inflater.inflate(R.layout.content_main, view, false);
+    if (KiwixApplication.getInstance().getResources().getConfiguration().orientation ==
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: It is not so difficult to use a landscape layout in android. You can read this to get it done easily: https://stackoverflow.com/questions/2124046/how-do-i-specify-different-layouts-for-portrait-and-landscape-orientations.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java [hunk 4]
```diff
@@ -106,11 +112,34 @@ public void onPageFinished(WebView view, String url) {
...
+    }
     callback.setHomePage(home);
     if (sharedPreferenceUtil.nightMode()) {
       ImageView cardImage = home.findViewById(R.id.content_main_card_image);
-      cardImage.setImageResource(R.drawable.ic_home_kiwix_banner_night);
+      ImageView sideImage = home.findViewById(R.id.content_side_image);
+      AppCompatButton downloadButton = home.findViewById(R.id.content_main_card_download_button);
+      downloadButton.setTextColor(Color.parseColor("#000000"));
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+        downloadButton.setBackgroundTintList(ColorStateList.valueOf(
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: You can do this using Butter Knife. You can refer to http://jakewharton.github.io/butterknife/ and find **R.color** on this webpage to search how to inject color in an instance variable.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 5]
```diff
@@ -1343,7 +1343,7 @@ public void openMainPage() {
 
   private void setUpWebView() {
     getCurrentWebView().getSettings().setJavaScriptEnabled(true);
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: I think you should remove the static import of `VERSION_CODES.KITKAT` and keep this line as it is. This will save us 1 line and it is more readable.

File: app/src/main/res/layout/content_main_p.xml [hunk 6]
```diff
@@ -20,19 +20,29 @@
     <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        >
+        android:id="@+id/constraint_main"
+        android:background="@drawable/back_cover">
```
Comment (abdulwd) [inline app/src/main/res/layout/content_main_p.xml]: You have not formatted all your files using Square's code style as mentioned in the repo's docs. You just need to press `Ctrl + Shift + L` to autoformat all the lines.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java (hunk 4 above)
Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: I tried doing using Butterknife but it is throwing an error " Resource not found exception" so I did this.

**Comment by sonusourav** [comment]
@abdulwd @siddharth2010  I have made some changes. Please review it.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 7]
```diff
@@ -1015,7 +1015,7 @@ private void openZimFile(File file, boolean clearHistory) {
       ActivityCompat.requestPermissions(this,
           new String[] { Manifest.permission.READ_EXTERNAL_STORAGE },
           REQUEST_STORAGE_PERMISSION);
-      if (BuildConfig.IS_CUSTOM_APP && Build.VERSION.SDK_INT == Build.VERSION_CODES.M) {
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: You should revert this change and delete this line `import static android.os.Build.VERSION_CODES;` so that there is consistency in using version codes.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -243,4 +243,5 @@
   <string name="switch_tabs">Switch tabs</string>
   <string name="smiling_face">:D</string>
   <string name="close_all_tabs">Close all tabs</string>
+  <string name="side_design">sideDesign</string>
```
Comment (abdulwd) [inline app/src/main/res/values/strings.xml]: I cannot find the use of this line.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java [hunk 9]
```diff
@@ -108,9 +114,30 @@ private void inflateHomeView(WebView view) {
...
+            KiwixApplication.getInstance().getResources().getColor(R.color.complement_blue800)));
+      }
+
+      ConstraintLayout constraintLayout = home.findViewById(R.id.constraint_main);
+      constraintLayout.setBackgroundResource(R.drawable.back_cover_night);
+      sideImage.setImageResource(R.drawable.home_side_cover_night);
+
+      if (KiwixApplication.getInstance().getResources().getConfiguration().orientation ==
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: Earlier you were using layout resource like this. You can also do the same with `drawables` or any other resources. You can read this https://developer.android.com/guide/topics/resources/providing-resources for more info.

File: app/src/main/res/values/strings.xml (hunk 8 above)
Comment (sonusourav) [reply app/src/main/res/values/strings.xml]: Line removed.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java (hunk 7 above)
Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: Well, this line was there from before I cloned this repo. You can check that.

**Comment by sonusourav** [comment]
@abdulwd  @siddharth2010  Please review these once.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java [hunk 10]
```diff
@@ -108,9 +114,30 @@ private void inflateHomeView(WebView view) {
...
+          Configuration.ORIENTATION_PORTRAIT) {
+        cardImage.setImageResource(R.drawable.kiwix_vertical_logo_night);
+      } else if (KiwixApplication.getInstance().getResources().getConfiguration().orientation ==
+          Configuration.ORIENTATION_LANDSCAPE) {
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: As I have told you before you should not check the orientation dynamically. Instead you should have the same name of the drawables for both the configuration and move the drawables for landscape to folders `drawable-land-hdpi`, `drawable-land-mdpi`, etc.

Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: @abdulwd Made all the changes.

**Comment by sonusourav** [comment]
@abdulwd Changes done.

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java [hunk 11]
```diff
@@ -108,9 +114,21 @@ private void inflateHomeView(WebView view) {
...
     home = inflater.inflate(R.layout.content_main, view, false);
     callback.setHomePage(home);
+
     if (sharedPreferenceUtil.nightMode()) {
       ImageView cardImage = home.findViewById(R.id.content_main_card_image);
-      cardImage.setImageResource(R.drawable.ic_home_kiwix_banner_night);
+      ImageView sideImage = home.findViewById(R.id.content_side_image);
+      AppCompatButton downloadButton = home.findViewById(R.id.content_main_card_download_button);
+      ConstraintLayout constraintLayout = home.findViewById(R.id.constraint_main);
+      constraintLayout.setBackgroundResource(R.drawable.back_cover_night);
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: A very nice follow-up. I think you can make use of this answer https://stackoverflow.com/a/13735919/7792129 to change the image based on the theme. Although we were doing it dynamically we should change it. This will move the UI implementation to the xml where it should be.

Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java]: @abdulwd I tried this but it's not working. The app crashes when I did that.
Also, I don't think it's worth adding 4 lines when we are getting the same desired result in 1 line.

**Comment by sonusourav** [comment]
@abdulwd @siddharth2010 @mhutti1 Do I have to make any other changes?

**Comment by abdulwd** [comment]
@sonusourav It will be great if you can rebase your branch and squash all your commit into a single commit. PR becomes easy to review.
